staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/controller.go

```diff
@@ -170,70 +171,79 @@ func (c *Controller) sync(name string) error {
 		delete(c.crdSpecs, name)
 		klog.V(2).Infof("Updating CRD OpenAPI spec because %s was removed", name)
 		regenerationCounter.With(map[string]string{"crd": name, "reason": "remove"})
-		return c.updateSpecLocked()
+		return c.buildMergeSpecLocked()
 	}
 
 	// compute CRD spec and see whether it changed
-	oldSpecs, updated := c.crdSpecs[crd.Name]
-	newSpecs, changed, err := buildVersionSpecs(crd, oldSpecs)
+	oldCRD, exists := c.crdSpecs[crd.Name]
+	if exists {
+		for _, versionCRD := range oldCRD {
+			c := versionCRD.Get()
```

Comment:
is this code for debugging purposes? I'm concerned about the `Get` here.

Reply:
`Get()` is thread safe now so this would purely be a performance concern? If the controller deems that the CRD needs to be synced, we need to determine whether the CRD has changed and needs to be replaced. I think `Get()` serves that purpose to fetch the ETag. This technically also helps with warming the cache if `Get()` has never been called before.

Originally I thought about always replacing regardless, but that would be suboptimal in the case that the CRD spec has already been constructed and we need to reconstruct on an overwrite.

Reply:
Discussed offline, will find a way to do this without calling `Get()`